File: sdk/npm/examples/json-bus.mjs
import { mkdtempSync, rmSync } from "node:fs";
import { tmpdir } from "node:os";
import { join } from "node:path";
import { AgoraClient } from "../dist/index.js";

const home = mkdtempSync(join(tmpdir(), "agora-node-sdk-example-"));

try {
  const agora = new AgoraClient({
    home,
    agentId: "node-sdk-example",
    relayUrl: "memory://node-sdk-example",
  });

  const room = await agora.joinRoom(
    "ag-node-sdk-example",
    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "example-bus"
  );

  await room.sendJson({
    kind: "job",
    id: "job-42",
    body: { command: "summarize", path: "README.md" },
  });

  const frames = await room.fetchJson({ limit: 10 });
  for (const frame of frames) {
    console.log(`${frame.agentId} sent ${frame.value.kind}:${frame.value.id}`);
  }
} finally {
  rmSync(home, { recursive: true, force: true });
}

File: sdk/python/examples/json_bus.py
"""Send and read application JSON frames over an Agora room.

Set AGORA_ROOM_ID and AGORA_ROOM_SECRET to an existing room before running:

    AGORA_ROOM_ID=ag-... AGORA_ROOM_SECRET=... python examples/json_bus.py
"""

from __future__ import annotations

import os
import tempfile

from agora_chat import AgoraClient


def main() -> None:
    room_id = os.environ["AGORA_ROOM_ID"]
    secret = os.environ["AGORA_ROOM_SECRET"]

    client = AgoraClient(
        home=os.environ.get("AGORA_HOME") or tempfile.mkdtemp(prefix="agora-python-sdk-"),
        agent_id=os.environ.get("AGORA_AGENT_ID", "python-sdk-example"),
        relay_url=os.environ.get("AGORA_RELAY_URL"),
        relay_token=os.environ.get("AGORA_RELAY_TOKEN"),
    )
    room = client.join_room(room_id, secret, label="example-bus")

    room.send_json(
        {
            "kind": "job",
            "id": "job-42",
            "body": {"command": "summarize", "path": "README.md"},
        }
    )

    for event in room.fetch_json(since="10m"):
        print(f"{event.message.sender} sent {event.value['kind']}:{event.value['id']}")


if __name__ == "__main__":
    main()
